src/main/java/uk/gov/justice/hmpps/offenderevents/services/ExternalApiService.java

```diff
@@ -1,34 +1,47 @@
 package uk.gov.justice.hmpps.offenderevents.services;
 
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.core.ParameterizedTypeReference;
-import org.springframework.http.HttpMethod;
-import org.springframework.security.oauth2.client.OAuth2RestTemplate;
 import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
 import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.justice.hmpps.offenderevents.config.OffenderEventsProperties;
 import uk.gov.justice.hmpps.offenderevents.model.OffenderEvent;
 
 import java.time.LocalDateTime;
 import java.util.List;
 
+import static org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction.clientRegistrationId;
+
 @Service
 @AllArgsConstructor
+@Slf4j
 public class ExternalApiService {
 
-    private final OAuth2RestTemplate custodyapiRestTemplate;
+    private final OffenderEventsProperties properties;
+    private final WebClient oauth2WebClient;
 
     public List<OffenderEvent> getEvents(final LocalDateTime fromDateTime, final LocalDateTime toDateTime) {
         final var uri = getEventUriBuilder(fromDateTime, toDateTime).build().toUri();
-        final var response = custodyapiRestTemplate.exchange(uri, HttpMethod.GET, null, new ParameterizedTypeReference<List<OffenderEvent>>() {
-        });
-        return response.getBody();
+        try {
+            return oauth2WebClient.get()
+                    .uri(uri)
+                    .attributes(clientRegistrationId("custody-api"))
+                    .retrieve()
+                    .bodyToMono(new ParameterizedTypeReference<List<OffenderEvent>>() {
+                    })
+                    .doOnError(error -> log.error("Error occurred", error))
+                    .block();
```

Comment:
has the potential to block forever - should generally pass a time out to these kind of things